tensorflow/models/rnn/ptb/ptb_word_lm.py

```diff
@@ -239,26 +239,39 @@ class TestConfig(object):
   vocab_size = 10000
 
 
-def run_epoch(session, m, data, eval_op, verbose=False):
+def run_epoch(session, model, data, eval_op, verbose=False):
   """Runs the model on the given data."""
-  epoch_size = ((len(data) // m.batch_size) - 1) // m.num_steps
+  epoch_size = ((len(data) // model.batch_size) - 1) // model.num_steps
   start_time = time.time()
   costs = 0.0
   iters = 0
-  state = m.initial_state.eval()
-  for step, (x, y) in enumerate(reader.ptb_iterator(data, m.batch_size,
-                                                    m.num_steps)):
-    cost, state, _ = session.run([m.cost, m.final_state, eval_op],
-                                 {m.input_data: x,
-                                  m.targets: y,
-                                  m.initial_state: state})
+  state = []
+  for c, m in model.initial_state: # initial_state: ((c1, m1), (c2, m2))
+    state.append((c.eval(), m.eval()))
+  for step, (x, y) in enumerate(reader.ptb_iterator(data, model.batch_size,
+                                                    model.num_steps)):
+    fetches = []
+    fetches.append(model.cost)
```

Comment:
Looks like you have to feed those separately.

On Jul 28, 2016 3:47 AM, "Dipanjan Sarkar" <notifications@github.com> wrote:

> In tensorflow/models/rnn/ptb/ptb_word_lm.py
> <https://github.com/tensorflow/tensorflow/pull/2580#discussion_r72601158>:
>
> >    start_time = time.time()
> >    costs = 0.0
> >    iters = 0
> > -  state = m.initial_state.eval()
> > -  for step, (x, y) in enumerate(reader.ptb_iterator(data, m.batch_size,
> > -                                                    m.num_steps)):
> > -    cost, state, _ = session.run([m.cost, m.final_state, eval_op],
> > -                                 {m.input_data: x,
> > -                                  m.targets: y,
> > -                                  m.initial_state: state})
> > +  state = []
> > +  for c, m in model.initial_state: # initial_state: ((c1, m1), (c2, m2))
> > +    state.append((c.eval(), m.eval()))
> > +  for step, (x, y) in enumerate(reader.ptb_iterator(data, model.batch_size,
> > +                                                    model.num_steps)):
> > +    fetches = []
> > +    fetches.append(model.cost)
>
> Hi @ebrevdo <https://github.com/ebrevdo> ,
>
> Thanks for the suggestion, this works till the point where I am having
> some issues with the state, since we have m.initial_state: state in
> feed_dict but now our initial state is no longer a concatenated tensor
> but two different ones for the c and h states so how can we pass it to
> the feed_dict?
>
> I just started with TensorFlow recently so I might be missing something
> obvious.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/2580/files/2819a680c951220d30b47b2212535412a66bbe7a#r72601158>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABtimzBJOGYRRBqB4_F3hyBrqinFmwr9ks5qaIjGgaJpZM4IpzVZ>
> .
>